pkg/agent/agent.go

```diff
@@ -153,6 +153,10 @@ func FlowsAgent(cfg *Config) (*Flows, error) {
 		PromConnectionInfo: metrics.PromConnectionInfo{
 			Address: cfg.MetricsServerAddress,
 			Port:    cfg.MetricsPort,
+			TLS: &metrics.PromTLS{
```

Comment:
defined as such, `PromConnectionInfo.TLS` will never be nil; however, below you're checking with 
```
if conn.TLS != nil {
			err = httpServer.ListenAndServeTLS(conn.TLS.CertPath, conn.TLS.KeyPath)
}
```

So I guess you should check for empty `cfg.MetricsTLSCertPath` instead?

Reply:
fixed it